executor/src/witgen/processor.rs

```diff
@@ -154,6 +200,63 @@ impl<'a, 'b, T: FieldElement> Processor<'a, 'b, T> {
                 e
             })?;
 
+        Ok(self.apply_updates(row_index, &updates, || identity.to_string()))
+    }
+
+    fn process_outer_query(
+        &mut self,
+        row_index: usize,
+    ) -> Result<(bool, Constraints<&'a PolynomialReference, T>), EvalError<T>> {
+        let Calldata { left, right } = self
+            .calldata
+            .as_mut()
+            .expect("Asked to process outer query, but no calldata was set!");
```

Comment:
We could encode the fact if we have calldata into the (generic) type of Processor. Then the return type of `finish()` could also change (I think the second part of the tuple is None if we do not have calldata and the new values for the input if we do, right?).

But could also be overkill.